agent.go

```diff
@@ -157,7 +169,7 @@ type agent interface {
 	createPod(pod *Pod) error
 
 	// exec will tell the agent to run a command in an already running container.
-	exec(pod *Pod, c Container, process Process, cmd Cmd) error
+	exec(pod *Pod, c Container, process Process, cmd Cmd) (guestProcess, error)
```

Comment:
The change is okay but this means that we'll have to change the sequence from the caller, because right now, we're starting the shim first and calling into `agent.exec()` as a second step.
Also, I am worried about the `cc-shim` case where we were expecting the shim to be connected so that we won't miss any output. I am not sure if this will cause any trouble.